dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorArm8.java

```diff
@@ -57,7 +57,7 @@ public List<Event> visitInitLock(InitLock e) {
 
     @Override
     public List<Event> visitLock(Lock e) {
-        IntegerType type = types.getArchType();
+        IntegerType type = (IntegerType)e.getAccessType();
```

Comment:
can we use `BooleanType`? ... the same comment applies to all lock events touched in this PR